### PR TITLE
Update JWT documentation to recommend only using jwt_header or audit logging not both 

### PR DESCRIPTION
**Pull request by scrawfor99**
### Description
Because of the interaction between the audit logging system and custom REST headers, it is recommended users only use _either_ custom JWT headers or audit logging; not both. If a user configures a custom JWT header other than `Authorization` the audit log may record the JWT passed in requests. This can then be parsed from the logs or potentially re-used depending on the user's construction of the token. 

A long term fix for this scenario will be pursued but this change should be made to notify users in the meantime. 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by hdhalter** [comment]
Thanks, @scrawfor99 ! Is this ready for doc review? And does it have an associate issue?

**Comment by scrawfor99** [comment]
Hi @hdhalter it should be all set for doc review. This is the associated issue: https://github.com/opensearch-project/security/issues/3875

**Comment by hdhalter** [comment]
Thanks, @scrawfor99 - Is backporting to 2.11 and 1.3 appropriate?

**Comment by scrawfor99** [comment]
Hi @hdhalter yep!

File: _security/authentication-backends/jwt.md [hunk 1]
```diff
@@ -116,7 +116,7 @@ The following table lists the configuration parameters.
 Name | Description
 :--- | :---
 `signing_key` | The signing key to use when verifying the token. If you use a symmetric key algorithm, it is the base64-encoded shared secret. If you use an asymmetric algorithm, it contains the public key.
-`jwt_header` | The HTTP header in which the token is transmitted. This is typically the `Authorization` header with the `Bearer` schema: `Authorization: Bearer <token>`. Default is `Authorization`.
+`jwt_header` | The HTTP header in which the token is transmitted. This is typically the `Authorization` header with the `Bearer` schema: `Authorization: Bearer <token>`. Default is `Authorization`. NOTE: Replacing this field with a value other than `Authorization` prevents the audit log from properly redacting the JWT from audit messages. It is reccomended that users only use the standard `Authorization` `jwt_header` when using JWTs with audit logging.
```
Comment (Naarcha-AWS) [inline _security/authentication-backends/jwt.md:119]:
```suggestion
`jwt_header` | The HTTP header in which the token is transmitted. This is typically the `Authorization` header with the `Bearer` schema,`Authorization: Bearer <token>`. Default is `Authorization`. Replacing this field with a value other than `Authorization` prevents the audit log from properly redacting the JWT header from audit messages. It is recommended that users only use  `Authorization` when using JWTs with audit logging. 
```

**Review by Naarcha-AWS — COMMENTED** [review]
Make sure that my changes retain the intended meaning.

**Review by Naarcha-AWS — APPROVED** [review]

**Pull request merged**